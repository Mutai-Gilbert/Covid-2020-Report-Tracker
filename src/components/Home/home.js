import PropTypes from 'prop-types';
import { faArrowRight } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

const Homepage = ({ country, cases }) => (
  <div className="">
    <div className="homeinfo">
      <div className="homeinfo1">
        <i className="arrow">
          <i>
            <FontAwesomeIcon icon={faArrowRight} />
          </i>
        </i>
        <p className="country-section-text">
          {country}
        </p>
        <p className="cases">{cases}</p>
      </div>
    </div>
  </div>
);

Homepage.propTypes = {
  country: PropTypes.string.isRequired,
  cases: PropTypes.string.isRequired,
};

export default Homepage;
