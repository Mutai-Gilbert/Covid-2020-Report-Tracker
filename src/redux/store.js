import { configureStore, combineReducers, applyMiddleware } from '@reduxjs/toolkit';
import thunk from 'redux-thunk';
import covidSlice from './Diseases/diseasesSlice';

const reducers = combineReducers({
  covidTracker: covidSlice,
});

const store = configureStore({
  reducer: reducers,
}, applyMiddleware(thunk));

export default store;
