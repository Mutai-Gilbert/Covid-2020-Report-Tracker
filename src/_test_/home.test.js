import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import store from '../redux/store';
import Home from '../components/Home/home';

describe('Home component testing', () => {
  it('renders without crashing', () => {
    const home = render(
      <BrowserRouter>
        <Provider store={store}>
          <Home />
        </Provider>
      </BrowserRouter>,
    );
    expect(home).toMatchSnapshot();
  });
});
