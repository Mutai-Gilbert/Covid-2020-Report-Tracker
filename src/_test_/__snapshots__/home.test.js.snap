// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component testing renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class=""
      >
        <div
          class="homeinfo"
        >
          <div
            class="homeinfo1"
          >
            <i
              class="arrow"
            >
              <i>
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-arrow-right "
                  data-icon="arrow-right"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M438.6 278.6c12.5-12.5 12.5-32.8 0-45.3l-160-160c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L338.8 224 32 224c-17.7 0-32 14.3-32 32s14.3 32 32 32l306.7 0L233.4 393.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0l160-160z"
                    fill="currentColor"
                  />
                </svg>
              </i>
            </i>
            <p
              class="country-section-text"
            />
            <p
              class="cases"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class=""
    >
      <div
        class="homeinfo"
      >
        <div
          class="homeinfo1"
        >
          <i
            class="arrow"
          >
            <i>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-arrow-right "
                data-icon="arrow-right"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M438.6 278.6c12.5-12.5 12.5-32.8 0-45.3l-160-160c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L338.8 224 32 224c-17.7 0-32 14.3-32 32s14.3 32 32 32l306.7 0L233.4 393.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0l160-160z"
                  fill="currentColor"
                />
              </svg>
            </i>
          </i>
          <p
            class="country-section-text"
          />
          <p
            class="cases"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
